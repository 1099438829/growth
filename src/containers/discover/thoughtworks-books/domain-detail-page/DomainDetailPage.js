import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ScrollView } from 'react-native';
import AppStyle from '../../../../theme/styles';
import TWBOOKS from '../../../../constants/TWBOOKS';
import PageItem from '../../../../components/discover/thought-works-books/view/DomainDetailPageItem';

function getBooks(domain) {
  switch (domain) {
    case 'practical':
      return TWBOOKS['zh-cn'].practical.books;
    case 'architecture':
      return TWBOOKS['zh-cn'].architecture.books;
    case 'methodology':
      return TWBOOKS['zh-cn'].methodology.books;
    case 'leadership':
      return TWBOOKS['zh-cn'].leadership.books;
    default:
      return [];
  }
}

class DomainDetailPage extends Component {
  static componentName = 'DomainDetailPage';

  static propTypes = {
    domain: PropTypes.string.isRequired,
  };

  static defaultProps = {
    domain: 'practical',
  };

  constructor(props) {
    super(props);
    this.state = {
      books: getBooks(this.props.domain),
      data: [
        { name: '基础篇', books: getBooks(this.props.domain).basic },
        { name: '进阶篇', books: getBooks(this.props.domain).intermediate },
        { name: '高级篇', books: getBooks(this.props.domain).advanced },
      ],
    };
  }

  render() {
    const rows = this.state.data.map((val, index) => <PageItem content={val} key={'key'.concat(index)} />);
    return <ScrollView style={AppStyle.detailBasisStyle}>{rows}</ScrollView>;
  }
}
export default DomainDetailPage;
