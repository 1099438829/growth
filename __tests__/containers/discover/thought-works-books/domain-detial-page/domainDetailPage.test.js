import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import DomainDetailPage from '../../../../../src/containers/discover/thoughtworks-books/domain-detail-page/DomainDetailPage';

it('renders correctly', () => {
  const domainDetailPage = renderer.create(
    <DomainDetailPage title={'实践'} domain={'practical'} />,
  );
  const domainDetailPageJson = domainDetailPage.toJSON();
  expect(domainDetailPageJson).toMatchSnapshot();
});
