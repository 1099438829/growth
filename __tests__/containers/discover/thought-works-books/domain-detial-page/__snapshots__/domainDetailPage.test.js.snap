// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E9EBEE",
      "marginTop": 64,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "rgb(245,245,245)",
          "height": 40,
          "justifyContent": "center",
          "paddingLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#999",
          }
        }
      >
        基础篇
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          代码整洁之道
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          单元测试之道
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          卓有成效的程序员
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          测试驱动开发
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          程序员的职业修养
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          编写可读代码的艺术
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "rgb(245,245,245)",
          "height": 40,
          "justifyContent": "center",
          "paddingLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#999",
          }
        }
      >
        进阶篇
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          重构与模式
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          实现模式
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          代码大全
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          程序员修炼之道
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "rgb(245,245,245)",
          "height": 40,
          "justifyContent": "center",
          "paddingLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#999",
          }
        }
      >
        高级篇
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          计算机程序的构造和解释
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          修改代码的艺术
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
